03week/towersOfHanoi.js

```diff
@@ -13,33 +13,68 @@ let stacks = {
   c: []
 };
 
-function printStacks() {
+const possibleInputs = ['a', 'b', 'c']
+
+const printStacks=()=> {
+  console.log("--------------");//added lines before and after stacks to increase readability
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
+  console.log("--------------");
 }
 
-function movePiece() {
-  // Your code here
-
+const towersOfHanoi=(startStack, endStack)=> {
+  isLegal(startStack, endStack);//triggers the isLegal() function to check for valid moves and input before moving pieces
 }
 
-function isLegal() {
-  // Your code here
-
+const isLegal=(startStack, endStack)=> {
+  //check for valid stack letter input of a, b or c. if input is not a,b,c
+  //it will have a -1 index due to not being in possibleInputs array
+  if (possibleInputs.indexOf(startStack) === -1 || possibleInputs.indexOf(endStack) === -1){
+    console.log("Not a valid stack input.")
+    return true
+  //prevent movePiece() from popping an empty string from startStack and pushing
+  //an empty string to endStack by checking array length
+  } else if (stacks[startStack].length === 0){
+    console.log("Can't start from an empty stack!")
+    return true
+  //prevent movePiece() from adding a greater number to a stack/array ending with a larger number
+  } else if (stacks[startStack][stacks[startStack].length-1] > stacks[endStack][stacks[endStack].length-1]){
+    console.log("Can't move larger number on top of smaller number")
+    return true
+  //if move is legal then it will return the movePiece() function
+  } else {
+    return movePiece(startStack, endStack);
+  }
 }
 
-function checkForWin() {
-  // Your code here
-
+const movePiece=(startStack, endStack)=> {
+  //use pop() method to remove last item from startStack array and use push() method
+  //to add the popped number to endStack array
+  stacks[endStack].push(stacks[startStack].pop())
+  //check for win after every move
+  return checkForWin();
 }
 
-function towersOfHanoi(startStack, endStack) {
-  // Your code here
+const checkForWin=()=> {
+  if (stacks.b.length === 4 || stacks.c.length === 4){//checks for the only 2 cases of winning
+    console.log("You're a winner!!")
+    printStacks();//show the winner their winning stack before resetting the game board
```

Comment:
This printStacks() should be called from the towersOfHanoi() function. The only responsibility of checkForWin is to move the piece, not to return the return of checkForWin()